### DISCUSSION TITLE
Error with cg_poly_elements_read

### DISCUSSION THREAD

**Discussion by davetaflin** [CGNS/CGNS#286] [CGNStalk]
A customer is trying to open the attached file (output by a commercial CFD code) in Tecplot and encountering an error. I observe that cg_poly_elements_read is not returning anything to array ConnectOffset. This seems similar to behavior I saw here: https://cgnsorg.atlassian.net/browse/CGNS-219?jql=text%20~%20%22cg_poly_elements_read%22

cgnscheck reports 2 errors on this file:

ERROR:number 3D elements < specified in zone dimensions
ERROR:no 3D elements exist and CellDim is 3

Is there anything else I can try or advise the customer to try, or is this just a pathological file and the customer's solver needs to be fixed? I'd appreciate any insights or suggestions, thanks.

Dave
 
[Case-242.cgns.tar.gz](https://github.com/CGNS/CGNS/files/6743144/Case-242.cgns.tar.gz)

**Comment by MicK7** [comment]
The CGNSLibraryVersion_t node of the file is not correct (following bug https://cgnsorg.atlassian.net/browse/CGNS-236). The customer should upgrade its CGNS library to 3.4.2.

Concerning the ERROR reported by cgnscheck, it is legitimate as the specific case of shell cells was not foreseen when cgnscheck was coded.

A quick fix is to modify manually in cgnsview the CGNSLibrary_t node to set it to 3.3
